ts/features/bonus/cgn/screens/merchants/CgnMerchantSearchScreen.tsx

```diff
@@ -0,0 +1,315 @@
+    () => ({
+      ...IOStyles.horizontalContentPadding,
+      marginTop: insets.top,
+      paddingVertical: INPUT_PADDING
+    }),
+    [insets.top]
+  );
+
+  const merchantsRemoteValue = useIOSelector(cgnSearchMerchantsSelector);
+  const isSearchActive =
+    searchTextDebouncedTrimmed.length >= MIN_SEARCH_TEXT_LENGTH;
+  const merchants = isSearchActive ? getValue(merchantsRemoteValue) : undefined;
+  useEffect(() => {
+    if (isSearchActive) {
+      dispatch(
+        cgnSearchMerchants.request({
+          token: searchTextDebouncedTrimmed as NonEmptyString
+        })
+      );
+    }
+  }, [dispatch, isSearchActive, searchTextDebouncedTrimmed]);
+
+  const renderItemCallback = useCallback(
+    (itemInfo: ListRenderItemInfo<SearchItem>) => (
+      <MerchantSearchResultListItem
+        item={itemInfo.item}
+        searchText={searchTextDebouncedTrimmed}
+      />
+    ),
+    [searchTextDebouncedTrimmed]
+  );
+
+  const merchantsCountRemoteValue = useIOSelector(cgnMerchantsCountSelector);
+  useEffect(() => {
+    dispatch(cgnMerchantsCount.request());
+  }, [dispatch]);
+  const merchantsCount = getValue(merchantsCountRemoteValue);
+
+  const renderListEmptyComponent = useCallback(() => {
+    if (searchText.trim().length < MIN_SEARCH_TEXT_LENGTH) {
+      return (
+        <EmptyList
+          pictogram="searchLens"
+          title={I18n.t("bonus.cgn.merchantSearch.emptyList", {
+            merchantCount: merchantsCount
+          })}
+        />
+      );
+    }
+
+    return null;
+  }, [merchantsCount, searchText]);
+
+  const handleCancel = useCallback(() => {
+    navigation.goBack();
+  }, [navigation]);
+
+  useFocusEffect(
+    useCallback(() => {
+      searchInputRef.current?.focus();
+    }, [])
+  );
+
+  return (
+    <>
+      <View style={containerStyle}>
+        <SearchInput
+          accessibilityLabel={I18n.t(
+            "bonus.cgn.merchantSearch.input.placeholder"
+          )}
+          autoFocus={true}
+          cancelButtonLabel={I18n.t("bonus.cgn.merchantSearch.input.cancel")}
+          clearAccessibilityLabel={I18n.t(
+            "bonus.cgn.merchantSearch.input.clear"
+          )}
+          keepCancelVisible={true}
+          onCancel={handleCancel}
+          onChangeText={setSearchText}
+          placeholder={I18n.t("bonus.cgn.merchantSearch.input.placeholder")}
+          ref={searchInputRef}
+          value={searchText}
+        />
+      </View>
+      <FlatList
+        ItemSeparatorComponent={() => <Divider />}
+        data={merchants?.length !== 0 ? merchants : undefined}
+        keyExtractor={item => item.id}
+        renderItem={renderItemCallback}
+        ListEmptyComponent={renderListEmptyComponent}
+        keyboardDismissMode={Platform.select({
+          ios: "interactive",
+          default: "on-drag"
+        })}
+        keyboardShouldPersistTaps="handled"
+      />
+    </>
+  );
+}
+
+function MerchantSearchResultListItem({
```

Comment:
This component is only 50LOC, and is not used anywhere else. In my opinion it would be an unnecessary indirection. I can't picture two separate change requests where one would affect only the list item component. I would have put it inline if it wasn't for flatlist performance purposes.
Let me know how you would like to proceed

Reply:
Even though this component is only 50 lines of code, it’s still beneficial to keep it separate. This approach can be useful for testing purposes and allows for potential reuse if needed in the future. Additionally, maintaining it as an independent component makes future maintenance easier by focusing on a limited scope, rather than embedding it within a file that handles other responsibilities.